Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Rpow.lean

```diff
@@ -0,0 +1,333 @@
+  simp only [sqrt, nnrpow, NNReal.coe_inv, NNReal.coe_ofNat, NNReal.rpow_eq_pow]
+  congr
+  ext
+  exact_mod_cast NNReal.sqrt_eq_rpow _
+
+@[simp]
+lemma sqrt_zero : sqrt (0 : A) = 0 := by simp [sqrt]
+
+variable [UniqueNonUnitalContinuousFunctionalCalculus ℝ≥0 A]
+
+@[simp]
+lemma nnrpow_sqrt {a : A} {x : ℝ≥0} : (sqrt a) ^ x = a ^ (x / 2) := by
+  rw [sqrt_eq_nnrpow, nnrpow_nnrpow, one_div_mul_eq_div 2 x]
+
+lemma nnrpow_sqrt_two {a : A} (ha : 0 ≤ a := by cfc_tac) : (sqrt a) ^ (2 : ℝ≥0) = a := by
+  simp only [nnrpow_sqrt, ne_eq, OfNat.ofNat_ne_zero, not_false_eq_true, div_self]
+  rw [nnrpow_one]
+
+lemma sqrt_mul_sqrt_self {a : A} (ha : 0 ≤ a := by cfc_tac) : sqrt a * sqrt a = a := by
+  rw [← nnrpow_two, nnrpow_sqrt_two]
+
+@[simp]
+lemma sqrt_nnrpow {a : A} {x : ℝ≥0} : sqrt (a ^ x) = a ^ (x / 2) := by
+  simp [sqrt_eq_nnrpow, div_eq_mul_inv]
+
+lemma sqrt_nnrpow_two {a : A} (ha : 0 ≤ a := by cfc_tac) : sqrt (a ^ (2 : ℝ≥0)) = a := by
+  simp only [sqrt_nnrpow, ne_eq, OfNat.ofNat_ne_zero, not_false_eq_true, div_self]
+  rw [nnrpow_one]
+
+lemma sqrt_mul_self {a : A} (ha : 0 ≤ a := by cfc_tac) : sqrt (a * a) = a := by
+  rw [← nnrpow_two, sqrt_nnrpow_two]
+
+end sqrt
+
+end NonUnital
+
+section Unital
+
+variable {A : Type*} [PartialOrder A] [NormedRing A] [StarRing A] [StarOrderedRing A]
+  [TopologicalRing A] [NormedAlgebra ℂ A] [CompleteSpace A]
+  [ContinuousFunctionalCalculus ℝ≥0 (fun (a : A) => 0 ≤ a)]
+
+/- ## `rpow` -/
+
+/-- Real powers of operators, based on the unital continuous functional calculus. -/
+noncomputable def rpow (a : A) (y : ℝ) : A := cfc (fun x : ℝ≥0 => x ^ y) a
+
+/-- Enable `a ^ y` notation for `CFC.rpow`. This is a low-priority instance to make sure it does
+not take priority over other instances when they are available. -/
+noncomputable instance (priority := 100) : Pow A ℝ where
+  pow a y := rpow a y
+
+@[simp]
+lemma rpow_eq_pow {a : A} {y : ℝ} : rpow a y = a ^ y := rfl
+
+@[simp]
+lemma rpow_nonneg {a : A} {y : ℝ} : 0 ≤ a ^ y := cfc_predicate _ a
+
+lemma rpow_def {a : A} {y : ℝ} : a ^ y = cfc (fun x : ℝ≥0 => x ^ y) a := rfl
+
+lemma rpow_natCast {a : A} {n : ℕ} (ha : 0 ≤ a := by cfc_tac) : a ^ (n : ℝ) = a ^ n := by
+  have h₁ : (fun x : ℝ≥0 => x ^ (n : ℝ)) = fun (x : ℝ≥0) => x ^ n := by ext; simp
+  simp_rw [rpow_def, h₁]
+  change cfc (fun x : ℝ≥0 => (id x) ^ n) a = a ^ n
+  rw [cfc_pow (id : ℝ≥0 → ℝ≥0) n a, cfc_id ℝ≥0 a]
+
+@[simp]
+lemma rpow_algebraMap {x : ℝ≥0} {y : ℝ} :
+    (algebraMap ℝ≥0 A x) ^ y = algebraMap ℝ≥0 A (x ^ y) := by
+  simp only [rpow_def]
+  rw [cfc_algebraMap x (fun z : ℝ≥0 => z ^ y)]
+
+lemma rpow_add_of_zero_not_mem_spectrum {a : A} {x y : ℝ} (ha : 0 ∉ spectrum ℝ≥0 a) :
+    a ^ (x + y) = a ^ x * a ^ y := by
+  simp only [rpow_def, NNReal.rpow_eq_pow]
+  have h : ∀ r, ContinuousOn (fun z : ℝ≥0 => z ^ (r : ℝ)) (spectrum ℝ≥0 a) := by
+    intro r z hz
+    exact ContinuousAt.continuousWithinAt <| NNReal.continuousAt_rpow_const <| Or.inl <| by aesop
+  rw [← cfc_mul _ _ a (h x) (h y)]
```

Comment:
It's not that the proof is super long, but we should definitely have this lemma, and we can make the automation work for us:
```lean
@[fun_prop]
theorem _root_.NNReal.continuousOn_rpow_const_compl_zero {r : ℝ} :
    ContinuousOn (fun z : ℝ≥0 => z ^ r) {0}ᶜ :=
  fun _ h => ContinuousAt.continuousWithinAt <| NNReal.continuousAt_rpow_const (.inl h)

-- even though this follows from `ContinuousOn.mono` and the previous lemma, we include it for
-- automation purposes with `fun_prop`, because the side goal `0 ∉ s ∨ 0 ≤ r` is often easy to check
@[fun_prop]
theorem _root_.NNReal.continuousOn_rpow_const {r : ℝ} {s : Set ℝ≥0}
    (h : 0 ∉ s ∨ 0 ≤ r) : ContinuousOn (fun z : ℝ≥0 => z ^ r) s :=
  h.elim (fun _ ↦ ContinuousOn.mono (s := {0}ᶜ) (by fun_prop) (by aesop))
    (NNReal.continuous_rpow_const · |>.continuousOn)
```
and then we get:
```suggestion
  rw [← cfc_mul _ _ a]
```